modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/utils/getYearlyTerms.js

```diff
@@ -4,52 +4,69 @@
  */
 
 const getYearlyTerms = ({endDate, startDate}) => {
-	const yearStartDate = new Date(startDate).getFullYear();
-	const yearEndDate = new Date(endDate).getFullYear();
+	const subscriptionStartDate = new Date(startDate);
+	const subscriptionEndDate = new Date(endDate);
 
-	if (yearStartDate + 1 < yearEndDate) {
-		const yearDateSplitted = new Array(yearEndDate - yearStartDate + 1)
+	if (
+		subscriptionStartDate.getFullYear() + 1 <
+		subscriptionEndDate.getFullYear()
+	) {
+		let arraySize =
+			subscriptionEndDate.getFullYear() -
+			subscriptionStartDate.getFullYear();
+
+		if (subscriptionEndDate.getMonth() > subscriptionStartDate.getMonth()) {
+			arraySize = arraySize + 1;
+		}
+
+		if (
+			subscriptionEndDate.getMonth() === subscriptionStartDate.getMonth()
+		) {
+			if (
```

Comment:
double nesting ifs can be better written with two conditions in the first if statement

Reply:
I've sent the adjustments, thanks!